assertEquals doesn't correctly compare Blobs

**Describe the bug**

`assertEquals` doesn't consider value-equal `Blob`s to be equal.

**Steps to Reproduce**

Run the following code:

```javascript
import { assertEquals } from "jsr:@std/assert";
assertEquals(new Blob(["foo"]), new Blob(["foo"]));
```

**Expected behavior**

This should not throw an error. Instead, it throws `AssertionError: Values are not equal.`

**Environment**

- OS: macOS 15
- deno version: 2.1.0
- std version: 1.0.8

--- a/assert/equals.ts
+++ b/assert/equals.ts
@@ -15,12 +15,25 @@ import { AssertionError } from "./assertion_error.ts";
  * Type parameter can be specified to ensure values under comparison have the
  * same type.
  *
+ * Note: When comparing `Blob` objects, you should first convert them to
+ * `Uint8Array` using the `Blob.arrayBuffer()` method and then compare their
+ * contents.
+ *
  * @example Usage
  * ```ts ignore
  * import { assertEquals } from "@std/assert";
  *
  * assertEquals("world", "world"); // Doesn't throw
  * assertEquals("hello", "world"); // Throws
+ * ```
+ * @example Compare `Blob` objects
+ * ```ts ignore
+ * import { assertEquals } from "@std/assert";
+ *
+ * const bytes1 = await new Blob(["foo"]).bytes();
+ * const bytes2 = await new Blob(["foo"]).bytes();
+ *
+ * assertEquals(bytes1, bytes2);
  * ```
  *
  * @typeParam T The type of the values to compare. This is usually inferred.

--- a/assert/equals.ts
+++ b/assert/equals.ts
@@ -16,7 +16,7 @@ import { AssertionError } from "./assertion_error.ts";
  * same type.
  *
  * Note: When comparing `Blob` objects, you should first convert them to
- * `Uint8Array` using the `Blob.arrayBuffer()` method and then compare their
+ * `Uint8Array` using the `Blob.bytes()` method and then compare their
  * contents.
  *
  * @example Usage

